lib/Dialect/FIRRTL/FIRRTLOps.cpp

```diff
@@ -1769,10 +1769,94 @@ static void printImplicitSSAName(OpAsmPrinter &p, Operation *op,
     namesDisagree = true;
   }
 
-  if (namesDisagree)
-    p.printOptionalAttrDict(op->getAttrs());
-  else
-    p.printOptionalAttrDict(op->getAttrs(), {"name"});
+  SmallVector<StringRef, 2> elides;
+
+  if (!namesDisagree)
+    elides.push_back("name");
+
+  // Elide "annotations" if it doesn't exist or if it is empty
+  auto annotationsAttr = op->getAttrOfType<ArrayAttr>("annotations");
+  if (!annotationsAttr || annotationsAttr.size() == 0)
+    elides.push_back("annotations");
+
+  p.printOptionalAttrDict(op->getAttrs(), elides);
+}
+
+//===----------------------------------------------------------------------===//
+// Custom attr-dict Directive that Elides Annotations
+//===----------------------------------------------------------------------===//
+
+static ParseResult parseElideAnnotations(OpAsmParser &parser,
+                                         NamedAttrList &resultAttrs) {
+
+  if (parser.parseOptionalAttrDict(resultAttrs))
```

Comment:
I'd simplify this method to just `return `parser.parseOptionalAttrDict(resultAttrs);`

Reply:
Fixed all locations of this in the rebase.